project/frontend/src/components/ResultsMap.js

```diff
@@ -58,34 +67,71 @@ function MapContainer(props) {
     );
   };
 
+  /** Creates all the markers that are going to be displayed
+   *  on the screen. It creates the marker corresponding
+   *  to the restaurant at currentCardIndex, the 3 markers
+   *  before that and the 3 markers after that if they are
+   *  valid indicies in the restaurants list.
+   */
   const createMarkers = () => {
-    if (!restaurants) {
-      return;
+    if (restaurants.length === 0) {
+      return null;
     }
-    let markers = [];
-    const numOfMarkers = Math.min(restaurants.length, 4);
-    for (let i = 0; i < numOfMarkers; i++) {
-      const coords = restaurants[i].key.latLngCoords;
-      const numInList = (i + 1).toString();
-      markers.push(
-        <MarkerIcon
-          lat={coords.lat}
-          lng={coords.lng}
-          id={i}
-          aria-label={'Your #' + numInList + ' Match'}
-        />
-      );
+    const markers = [];
+    let coords = restaurants[currentCardIndex].key.latLngCoords;
+    markers.push(
+      <MarkerIcon
+        lat={coords.lat}
+        lng={coords.lng}
+        id={currentCardIndex}
+        aria-label={'Your #' + (currentCardIndex + 1) + ' Match'}
+      />
+    );
+    const numOfMarkers = Math.min(restaurants.length, 3);
+    // Initialized i as 1 instead of 0 because the marker
+    // at currentCardIndex was already pushed to markers.
+    for (let i = 1; i <= numOfMarkers; i++) {
+      const nextCardIndex = currentCardIndex + i;
+      if (isValidIndex(nextCardIndex)) {
+        coords = restaurants[nextCardIndex].key.latLngCoords;
+        markers.push(
+          <MarkerIcon
+            lat={coords.lat}
+            lng={coords.lng}
+            id={nextCardIndex}
+            aria-label={'Your #' + (nextCardIndex + 1) + ' Match'}
+          />
+        );
+      }
+      const previousCardIndex = currentCardIndex - i;
+      if (isValidIndex(previousCardIndex)) {
+        coords = restaurants[previousCardIndex].key.latLngCoords;
+        markers.push(
+          <MarkerIcon
+            lat={coords.lat}
+            lng={coords.lng}
+            id={previousCardIndex}
+            aria-label={'Your #' + (previousCardIndex + 1) + ' Match'}
+          />
+        );
+      }
     }
     return markers;
   };
 
+  /** Checks that the current index is a valid index for
+   * the restaurant matches list.
+   */
+  const isValidIndex = (index) => {
+    return index >= 0 && index < restaurants.length;
```

Comment:
Super nit but since this is just a one liner I think we can make it `const isValidIndex = (index) => index >= 0 && index < restaurants.length;`